docs/developer/extending/apps/architecture/apl.mdx

```diff
@@ -1,3 +1,109 @@
 ---
 title: APL - Auth Persistence Layer
 ---
+
+APL is the core component of communication between the Saleor and the App.
+
+During installation, Saleor sends AppToken, which the App uses to communicate with graphQL API. The App must save this token and target domain - but where?
+
+# What is APL
+
+APL is an abstraction on key/value persistence that holds domains (under which Saleor API is available) and tokens for communication.
+
+Saleor provides a simple persistence with App Metadata. Still, if App doesn't have a token and domain saved (which is typical for serverless architecture), it can't also call Saleor to reach Metadata. Hence need for simple, fast storage to save auth data.
+
+Saleor doesn't require App to use any specific technology for persistence. That's why there are many implementations of APL available in the app-sdk package. And writing a custom one is easy.
+
+# Communication diagram
+
+This diagram explains flow how APL is used:
+
+Following diagram describes how APL is used to store token received from Saleor during registration
```

Comment:
```suggestion
The following diagram describes how APL is used to store tokens received from Saleor during registration:
```

Is this redundant? Because we have:
> This diagram explains flow how APL is used:

just above.

Reply:
I think the latter should stay 👍